src/Relation/Nullary/Reflects.agda

```diff
@@ -43,54 +45,54 @@ data Reflects (A : Set a) : Bool → Set a where
 -- `ofʸ`), and `invert` strips off the constructor to just give either
 -- the proof of `A` or the proof of `¬ A`.
 
-of : ∀ {b} → if b then A else ¬ A → Reflects A b
-of {b = false} ¬a = ofⁿ ¬a
-of {b = true }  a = ofʸ a
+of : if x then A else ¬ A → Reflects A x
+of {x = false} ¬a = ofⁿ ¬a
+of {x = true }  a = ofʸ a
 
-invert : ∀ {b} → Reflects A b → if b then A else ¬ A
+invert : Reflects A x → if x then A else ¬ A
 invert (ofʸ  a) = a
 invert (ofⁿ ¬a) = ¬a
 
 ------------------------------------------------------------------------
 -- Interaction with negation, product, sums etc.
 
+infixr 1 _⊎-reflects_
+infixr 2 _×-reflects_ _→-reflects_
+
+T-reflects : ∀ x → Reflects (T x) x
+T-reflects true  = of _
+T-reflects false = of id
+
 -- If we can decide A, then we can decide its negation.
-¬-reflects : ∀ {b} → Reflects A b → Reflects (¬ A) (not b)
+¬-reflects : Reflects A x → Reflects (¬ A) (not x)
 ¬-reflects (ofʸ  a) = ofⁿ (_$ a)
 ¬-reflects (ofⁿ ¬a) = ofʸ ¬a
```

Comment:
See also #2149 which naturally could be merged after this, or as part of it: something to discuss?

Reply:
I think let's keep that as a separate PR?